src/main/java/prefs/Preferences.java

```diff
@@ -9,186 +15,279 @@
     }
 
     /**
-     * Initialises a Preferences instance which always creates its config file.
-     * This will overwrite it with a default configuration if it already exists.
+     * @param configDirectory The directory that the config files are held in.
+     * @param sessionConfigFileName The name of the session config file
+     * @param userConfigFileName The name of the user config file
+     * @param ignoreExisting True if existing config file is to be ignored.
      */
-    public static Preferences create(String configFileName) {
-        return new Preferences(configFileName, true);
+    private Preferences(String configDirectory,
+                        String sessionConfigFileName,
+                        String userConfigFileName,
+                        boolean ignoreExisting) {
+        this.configDirectory = configDirectory;
+        this.sessionConfigFileName = sessionConfigFileName;
+        this.userConfigFileName = userConfigFileName;
+
+        if (ignoreExisting) {
+            this.sessionConfig = createConfig(DEFAULT_FILE_CONTENTS, SessionConfig.class);
+            this.userConfig = createConfig(DEFAULT_FILE_CONTENTS, UserConfig.class);
+        } else {
+            this.sessionConfig = loadConfig(configDirectory, sessionConfigFileName, SessionConfig.class);
+            this.userConfig = loadConfig(configDirectory, userConfigFileName, UserConfig.class);
+        }
     }
 
-    public void saveGlobalConfig() {
-        fileHandler.saveGlobalConfig(global);
+    /**
+     * Creates a new Config instance
+     * @param contents The JSON representation of an contents of the Config object
+     * @param configClass The class of the Config instance
+     * @return The created Config instance
+     */
+    private <T> T createConfig(String contents, Class<T> configClass) {
+        return JsonHelper.fromJsonString(contents, configClass);
     }
 
-    private void initGlobalConfig() {
-        global = fileHandler.initGlobalConfig();
-    }
+    /**
+     * Loads a Config instance from a file on disk
+     * @param configDirectory The directory that the config file is held in.
+     * @param configFileName The filename of the config file on disk
+     * @param configClass The class of the config to load
+     * @return The loaded Config instance
+     */
+    private <T> T loadConfig(String configDirectory, String configFileName, Class<T> configClass) {
+        String fileContents = DEFAULT_FILE_CONTENTS;
+        if (FileHelper.isFileExists(configDirectory, configFileName)) {
+            try {
+                fileContents = FileHelper.getFileContents(configDirectory, configFileName);
+            } catch (IOException e) {
+                // if we can't read the file, just ignore the existing files
+                HTLog.error(logger, e);
+                logger.error(e.toString());
+                fileContents = DEFAULT_FILE_CONTENTS;
```

Comment:
Try to avoid setting the variable to same value twice. Perhaps not give it a value at line 110?